datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/config/DatasafeConfig.java

```diff
@@ -24,21 +41,30 @@
 @Configuration
 public class DatasafeConfig {
 
+    private static final Set<String> ALLOWED_TABLES = ImmutableSet.of("users", "private_profiles", "public_profiles");
+
+    private DatasafeProperties datasafeProperties;
+
+    @Autowired
+    private DFSConfig dfsConfig;
+
+    @Autowired
+    private StorageService storageService;
+
+    @Inject
+    DatasafeConfig(DatasafeProperties datasafeProperties) {
+        this.datasafeProperties = datasafeProperties;
+    }
+
     @Bean
     DFSConfig dfsConfig(DatasafeProperties properties) {
-        return new DefaultDFSConfig(new Uri(properties.getSystemRoot()), properties.getKeystorePassword());
+        return new DefaultDFSConfig(new Uri(properties.getS3Path()), properties.getKeystorePassword());
     }
 
-    /**
-     * @return S3 based storage service
-     */
     @Bean
-    StorageService storageService(AmazonS3 s3, DatasafeProperties properties) {
-        return new S3StorageService(
-                s3,
-                properties.getBucketName(),
-                Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors())
-        );
+    DFSConfig multiDfsConfig(DatasafeProperties properties) {
```

Comment:
There should be only one DFSConfig bean - conditionalOn

Reply:
done